Add KeyBy

thetnaingtn opened the pull request:
Add support for lodash's keyBy utility

samber commented on README.md:
````diff
@@ -535,6 +536,28 @@ m2 := lo.MapValues[int, int64, string](m, func(x int64, _ int) string {
 // map[int]string{1: "1", 2: "2", 3: "3"}
 ```
 
+### KeyValue
````
KeyValue -> KeyBy

samber commented on README.md:
```diff
@@ -78,6 +78,7 @@ Supported helpers for maps:
 - FromEntries
 - Assign (merge of maps)
 - MapValues
+- KeyBy
```
Could you move it into the "Slice" section, please?

Also from map.go to slice.go 🙏

samber commented on map.go:
```diff
@@ -70,3 +70,14 @@ func MapValues[K comparable, V any, R any](in map[K]V, iteratee func(V, K) R) ma
 	return result
 }
+
+func KeyBy[T any, V comparable](entries []T, iteratee func(T) V) map[V]T {
```
"V" to "K" please

It will be more understandable.

samber commented on map_test.go:
```diff
@@ -6,6 +6,7 @@ import (
 	"testing"
 
 	"github.com/stretchr/testify/assert"
+	"golang.org/x/tools/go/expect"
```
i'm pretty sure you can remove this line

samber replied:
i'm pretty sure you can remove this line

samber commented:
Oops, it is the same helper than lo.ToMap actually.

Can you rename ToMap to KeyBy please ?

thetnaingtn commented:
> Oops, it is the same helper than lo.ToMap actually.
> 
> Can you rename ToMap to KeyBy please ?

Sure thing. Now, I removed duplicate KeyBy and renamed ToMap utility to KeyBy.  @samber

samber commented:
Thanks for your help and sorry for the wrong name.

Let's merge ;)
